src/wp-includes/https-detection.php

```diff
@@ -137,3 +141,54 @@ function wp_cron_conditionally_prevent_sslverify( $request ) {
 	}
 	return $request;
 }
+
+/**
+ * Checks whether a given HTML string is likely an output from this WordPress site.
+ *
+ * This function attempts to check for various common WordPress patterns whether they are included in the HTML string.
+ * Since any of these actions may be disabled through third-party code, this function may also return null to indicate
+ * that it was not possible to determine ownership.
+ *
+ * @since 5.7.0
+ * @access private
+ *
+ * @param string $html Full HTML output string, e.g. from a HTTP response.
+ * @return bool|null True/false for whether HTML was generated by this site, null if unable to determine.
+ */
+function wp_is_owned_html_output( $html ) {
+	// 1. Check if HTML includes the site's Really Simple Discovery link.
+	if ( has_action( 'wp_head', 'rsd_link' ) ) {
+		ob_start();
+		rsd_link();
+
+		$pattern = ob_get_clean();
+		return false !== strpos( $html, $pattern );
```

Comment:
I have a concern about this. Namely, if `rsd_link()` outputs:

```html
<link rel="EditURI" type="application/rsd+xml" title="RSD" href="https://example.com/xmlrpc.php?rsd" />
```

Then if a plugin is manipulating the output in any way, then the check can fail. For example, the AMP plugin loads the HTML into a DOM document which will normalize this out to remove the self-closing slash:

```html
<link rel="EditURI" type="application/rsd+xml" title="RSD" href="https://example.com/xmlrpc.php?rsd">
```

Other plugins do other optimizations, like remove unnecessary attribute quotes, like so:

```html
<link rel=EditURI type="application/rsd+xml" title=RSD href="https://example.com/xmlrpc.php?rsd">
```

And that would cause a failure as well.

Therefore, I think instead of using output buffering to capture the needle and then look for it in the `$html` haystack, it would be more reliable to instead look just for a key bit of information that will be inside of the attribute, like `esc_url( site_url( 'xmlrpc.php?rsd', 'rpc' ) )`. 

The same goes for the following checks.

Reply:
@westonruter I've pushed a fix for this in https://github.com/WordPress/wordpress-develop/pull/568/commits/2756b9bf929f8e5126b744254723b92a20e708d2 - can you review?